### PR TITLE
[vcpkg baseline] Fix pbc build failure with x64-osx

### PR DESCRIPTION
**Pull request by PhoebeHui**
pbc requires gmp library on macos, add the dependency port.

Failures in CI pipeline:
```
checking for __gmpz_init in -lgmp... no
************************
gmp library not found
-e add its path to LDFLAGS
see ./configure --help
************************
```
Related PR https://github.com/microsoft/vcpkg/pull/12727

### CODE REVIEW

File: ports/pbc/CONTROL [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Source: pbc
-Version: 0.5.14-2
-Build-Depends: mpir (windows)
+Version: 0.5.14
+Port-Version: 3
+Build-Depends: mpir (windows), gmp(osx)
```
Comment (PhoebeHui) [inline ports/pbc/CONTROL:4]:
```suggestion
Build-Depends: mpir (windows), gmp(!windows)
```

**Comment by JackBoosY** [comment]
Another solution (using vcpkg_*_make): 
[pbc.zip](https://github.com/microsoft/vcpkg/files/5021068/pbc.zip)

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
@strega-nil, could you help revirew and merge this PR firstly?

**Comment by Neumann-A** [comment]
>Another solution (using vcpkg_*_make):

was also wondering why it is not using `vcpkg_configure_make` etc.

**Comment by strega-nil** [comment]
Thanks for fixing this phoebe :)

**Pull request merged**